### ISSUE TITLE
Auto-export does not support attachment export

### ISSUE THREAD

**Issue by pablomayobre** [retorquere/zotero-better-bibtex#3153]
### Debug log ID

4HNQ2X7M-refs-euc/7.0.5

### What happened?

Exporting with these options:

![Image](https://github.com/user-attachments/assets/025f8569-14d2-42e7-a300-512fa3944126)

Results in:

![Image](https://github.com/user-attachments/assets/b33b0798-e951-4dd1-a30b-e6ecac77a516)

My library is composed of PDF files ~and a single HTML entry from Zotero Connect~

**Comment by retorquere** [comment]
Are you running a flatpak or snap packaged Zotero by any chance? I'm seeing some errors in 4HNQ2X7M-refs-euc/7.0.5 that I'm not usually seeing. The export popup is missing translations and it should not have allowed you to check keepUpdated and Export files together. Auto-export indeed does not support attachment export.

**Comment by pablomayobre** [comment]
I followed the instructions from Zotero to install/move the tarball into /opt/zotero 

Ubuntu 24.10

**Comment by pablomayobre** [comment]
Is there any actionable I can take?

When you say it shouldn't allow me to select both, should I only tick keep updated?

What's the correct way to have a BibLatex file with my references and annotations?

Is the missing translations an issue? Should I install them or add them somehow?

**Comment by retorquere** [comment]
> Is there any actionable I can take?

Please send a new log. Was the previous log sent from a snap or flatpak install? 

> When you say it shouldn't allow me to select both, should I only tick keep updated?

You can, but the fact that it's not disabled means there is some error, and I don't know what further effects this error has. I want to fix this error. 

> What's the correct way to have a BibLatex file with my references and annotations?

You're doing things right, I just need to address the problem. For that I first need to establish what the error is. The previous log does not have sufficient information. 

> Is the missing translations an issue? Should I install them or add them somehow?

No, they're bundled, it's just not being applied.

**Comment by pablomayobre** [comment]
Again, no flatpak or snap

New log: 5CK2KQ42-refs-euc/7.0.5

**Comment by pablomayobre** [comment]
I also seem to be getting the same error as #895 after generating a .bib file

```
Error: Command failed: pandoc --bibliography="./Bibliografía.bib"  --csl=https://raw.githubusercontent.com/citation-style-language/styles/refs/heads/master/apa.csl
pandoc: ./Bibliografa.bib: withBinaryFile: invalid argument (cannot encode character '\65533')
```

[Bibliografía.bib.txt](https://github.com/user-attachments/files/18871576/Bibliografia.bib.txt)

**Comment by pablomayobre** [comment]
@retorquere why is this awaiting user feedback? Anything else you need from me?

**Comment by retorquere** [comment]
The bot was on the fritz, apologies

**Comment by retorquere** [comment]
> I also seem to be getting the same error as [#895](https://github.com/retorquere/zotero-better-bibtex/issues/895) after generating a .bib file
> 
> ```
> Error: Command failed: pandoc --bibliography="./Bibliografía.bib"  --csl=https://raw.githubusercontent.com/citation-style-language/styles/refs/heads/master/apa.csl
> pandoc: ./Bibliografa.bib: withBinaryFile: invalid argument (cannot encode character '\65533')
> ```
> 
> [Bibliografía.bib.txt](https://github.com/user-attachments/files/18871576/Bibliografia.bib.txt)

Unrelated

**Comment by retorquere** [comment]
> > I also seem to be getting the same error as [#895](https://github.com/retorquere/zotero-better-bibtex/issues/895) after generating a .bib file
> > ```
> > Error: Command failed: pandoc --bibliography="./Bibliografía.bib"  --csl=https://raw.githubusercontent.com/citation-style-language/styles/refs/heads/master/apa.csl
> > pandoc: ./Bibliografa.bib: withBinaryFile: invalid argument (cannot encode character '\65533')
> > ```

I can't replicate this. I'm getting

```

**Comment by retorquere** [comment]
Escaped the file you uploaded looks like this (with the escaped characters written as `\u[hex number]`:

```
@article{burkovAllModelsAre,
  title = {\u201C{{All}} Models Are Wrong, but Some Are Useful.\u201D \u2014 {{George Box}}},
  author = {Burkov, Andriy},
  langid = {english},
  file = {/home/pablo/Zotero/storage/HTB7MTD8/Burkov - \u201CAll models are wrong, but some are useful.\u201D \u2014 George Box.pdf}
}

@article{howardDeepLearningCoders,
  title = {Deep {{Learning}} for {{Coders}} with {{Fastai}} and {{PyTorch}}},
  author = {Howard, Jeremy and Gugger, Sylvain},
  langid = {english},
  file = {/home/pablo/Zotero/storage/E2RM99MK/Howard and Gugger - Deep Learning for Coders with Fastai and PyTorch.pdf}
}

@article{kinsleyNeuralNetworksScratch,
  title = {Neural {{Networks}} from {{Scratch}} in {{Python}}},
  author = {Kinsley, Harrison and Kukie\u0142a, Daniel},
  langid = {english},
  file = {/home/pablo/Zotero/storage/YBWSAFIC/Kinsley and Kukie\u0142a - Neural Networks from Scratch in Python.pdf}
}

@book{perrottaProgrammingMachineLearning2020,
  title = {Programming Machine Learning: From Coding to Deep Learning},
  shorttitle = {Programming Machine Learning},
  author = {Perrotta, Paolo},
  editor = {Dvorak, Katharine},
  date = {2020},
  publisher = {Pragmatic Bookshelf},
  location = {Raleigh, North Carolina},
  abstract = {You\'ve decided to tackle machine learning - because you\'re job hunting, embarking on a new project, or just think self-driving cars are cool. But where to start? It\'s easy to be intimidated, even as a software developer. The good news is that it doesn\'t have to be that hard. Master machine learning by writing code one line at a time, from simple learning programs all the way to a true deep learning system. Tackle the hard topics by breaking them down so they\'re easier to understand, and build your confidence by getting your hands dirty. Peel away the obscurities of machine learning, starting from scratch and going all the way to deep learning. Machine learning can be intimidating, with its reliance on math and algorithms that most programmers don\'t encounter in their regular work. Take a hands-on approach, writing the Python code yourself, without any libraries to obscure what\'s really going on. Iterate on your design, and add layers of complexity as you go. Build an image recognition application from scratch with supervised learning. Predict the future with linear regression. Dive into gradient descent, a fundamental algorithm that drives most of machine learning. Create perceptrons to classify data. Build neural networks to tackle more complex and sophisticated data sets. Train and refine those networks with backpropagation and batching. Layer the neural networks, eliminate overfitting, and add convolution to transform your neural network into a true deep learning system. Start from the beginning and code your way to machine learning mastery. What You Need: The examples in this book are written in Python, but don\'t worry if you don\'t know this language: you\'ll pick up all the Python you need very quickly. Apart from that, you\'ll only need your computer, and your code-adept brain},
  isbn = {978-1-68050-772-0},
  langid = {english},
  annotation = {OCLC: 1155169278},
  file = {/home/pablo/Zotero/storage/RG37D9XX/Perrotta - 2020 - Programming machine learning from coding to deep learning.pdf}
}

@book{raschkaMachineLearningPyTorch2022,
  title = {Machine {{Learning}} with {{PyTorch}} and {{Scikit-Learn}}: {{Develop}} Machine Learning and Deep Learning Models with {{Python}}},
  shorttitle = {Machine {{Learning}} with {{PyTorch}} and {{Scikit-Learn}}},
  author = {Raschka, Sebastian},
  with = {Liu, Yuxi and Mirjalili, Vahid and Dzhulgakov, Dmytro},
  date = {2022},
  edition = {1},
  publisher = {Packt Publishing Limited},
  location = {Birmingham},
  abstract = {bThis book of the bestselling and widely acclaimed Python Machine Learning series is a comprehensive guide to machine and deep learning using PyTorch\'s simple to code framework/bh4Key Features/h4ulliLearn applied machine learning with a solid foundation in theory/liliClear, intuitive explanations take you deep into the theory and practice of Python machine learning/liliFully updated and expanded to cover PyTorch, transformers, XGBoost, graph neural networks, and best practices/li/ulh4Book Description/h4Machine Learning with PyTorch and Scikit-Learn is a comprehensive guide to machine learning and deep learning with PyTorch. It acts as both a step-by-step tutorial and a reference you\'ll keep coming back to as you build your machine learning systems. Packed with clear explanations, visualizations, and examples, the book covers all the essential machine learning techniques in depth. While some books teach you only to follow instructions, with this machine learning book, we teach the principles allowing you to build models and applications for yourself. Why PyTorch? PyTorch is the Pythonic way to learn machine learning, making it easier to learn and simpler to code with. This book explains the essential parts of PyTorch and how to create models using popular libraries, such as PyTorch Lightning and PyTorch Geometric. You will also learn about generative adversarial networks (GANs) for generating new data and training intelligent agents with reinforcement learning. Finally, this new edition is expanded to cover the latest trends in deep learning, including graph neural networks and large-scale transformers used for natural language processing (NLP). This PyTorch book is your companion to machine learning with Python, whether you\'re a Python developer new to machine learning or want to deepen your knowledge of the latest developments.h4What you will learn/h4ulliExplore frameworks, models, and techniques for machines to \'learn\' from data/liliUse scikit-learn for machine learning and PyTorch for deep learning/liliTrain machine learning classifiers on images, text, and more/liliBuild and train neural networks, transformers, and boosting algorithms/liliDiscover best practices for evaluating and tuning models/liliPredict continuous target outcomes using regression analysis/liliDig deeper into textual and social media data using sentiment analysis/li/ulh4Who this book is for/h4If you have a good grasp of Python basics and want to start learning about machine learning and deep learning, then this is the book for you. This is an essential resource written for developers and data scientists who want to create practical machine learning and deep learning applications using scikit-learn and PyTorch.Before you get started with this book, you\'ll need a good understanding of calculus, as well as linear algebra},
  isbn = {978-1-80181-931-2 978-1-80181-638-0},
  langid = {english},
  pagetotal = {1},
  file = {/home/pablo/Zotero/storage/SSQ5WIAV/Raschka - 2022 - Machine Learning with PyTorch and Scikit-Learn Develop machine learning and deep learning models wi.pdf}
}

@article{tibshiraniValeriePatrickHastie,
  title = {Valerie and {{Patrick Hastie}}},
  author = {Tibshirani, Sami and Friedman, Harry},
  langid = {english},
  file = {/home/pablo/Zotero/storage/4EX943XW/Tibshirani and Friedman - Valerie and Patrick Hastie.pdf}
}
```

there is no unicode escape character (the `\65533` aka `U+FFFD` in the file, so if pandoc is seeing one, it is being introduced after BBT wrote the file.

**Comment by retorquere** [comment]
> 🤖 this is your friendly neighborhood build bot announcing [test build 7.0.5.7703](https://github.com/retorquere/zotero-better-bibtex/releases/download/builds/zotero-better-bibtex-7.0.5.7703.xpi) ("testcase")

This build adds logging for the export popup, please install it, try another export, then send a new debug log.

**Comment by pablomayobre** [comment]
I didn't get the error this time around and the checboxes worker properly unselecting the conflicting entries

5XRT5SK6-refs-euc/7.0.5.7703

**Comment by retorquere** [comment]
Ugh sorry about that bot. I need to fix that.

**Comment by pablomayobre** [comment]
> there is no unicode replacement character (the \65533 aka U+FFFD) in the file, so if pandoc is seeing one, it is being introduced after BBT wrote the file.

I later realized that the error was on Pandoc side because my file had an accent in the file name `Bibliografía.bib` and it was trying to find `Bibliografa.bib` which caused that error, not this plugin's fault!

The extension seems to be working now!

**Comment by retorquere** [comment]
That's good, but I didn't change anything.

**Comment by pablomayobre** [comment]
Let me know if I can help you debug anything else! I'm glad it's working now

Just in case these were the installers I used.

PS: Sending them so that you can check their SHA against the releases, I trust you won't install these!

**Comment by retorquere** [comment]
No worries. I'm sure you got the right download.
